src/test/groovy/org/prebid/server/functional/tests/privacy/AuctionGppSpec.groovy

```diff
@@ -0,0 +1,226 @@
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.ENFORCED
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.NOT_ENFORCED
+import static org.prebid.server.functional.util.privacy.TcfConsent.GENERIC_VENDOR_ID
+import static org.prebid.server.functional.util.privacy.TcfConsent.PurposeId.BASIC_ADS
+
+class AuctionGppSpec extends BaseSpec {
+
+    def "PBS should populate gdpr to 1 when regs.gdpr is not specified and gppSid contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def gppSidsId = [TCF_EU_V2.valueAsInt]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs = new Regs(gdpr: null, gppSid: gppSidsId)
+        }
+
+        when: "PBS processes auction request"
+        def bidResponse = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr and gppSid from request"
+        def resolvedRequest = bidResponse.ext.debug.resolvedRequest.regs
+        assert resolvedRequest.gdpr == 1
+        assert resolvedRequest.gppSid == gppSidsId
+    }
+
+    def "PBS should populate gdpr to 0 when regs.gdpr is not specified and gppSid not contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def gppSidsId = [PBSUtils.getRandomNumber(TCF_EU_V2.valueAsInt)]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs = new Regs(gdpr: null, gppSid: gppSidsId)
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr and gppSid from request"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == 0
+        assert bidderRequest.regs.gppSid == gppSidsId
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid contains 2 and regs.gdpr isn't 1"() {
+        given: "Default bid request with gppSid and gdpr"
+        def gdpr = 0
+        def gppSectionsId = [TCF_EU_V2.valueAsInt]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = gdpr
+            regs.gppSid = gppSectionsId
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP scope does not match TCF2 scope"]
+
+        and: "Bidder request should contain the same value as in request"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == gdpr
+        assert bidderRequest.regs.gppSid == gppSectionsId
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid not contains 2 and regs.gdpr isn't 0"() {
```

Comment:
This test is literally identical to the previous one. Use parametrization instead of duplicating the same code.